build/appveyor/build_mixxx.bat

```diff
@@ -1,61 +1,150 @@
+  echo.
+  echo Example: build_mixxx.bat x64 release c:\mixxx\environments\msvc15-static-x86-release
+  exit /b 1
+)
+
+set MACHINE_X86="%1" == "x86"
+set CONFIG_RELEASE="%2" == "release"
+set WINLIB_DIR=%3
+
+SET BIN_DIR=%WINLIB_DIR%\bin
+SET LIB_DIR=%WINLIB_DIR%\lib
+SET INCLUDE_DIR=%WINLIB_DIR%\include
+set QT_VERSION=4.8.7
+SET QTDIR=%WINLIB_DIR%\Qt-%QT_VERSION%
+
+if NOT EXIST "%BIN_DIR%\scons.py" (
 echo.
-echo "For example : .\build.bat x64 release"
-exit /B 1
+echo You need to obtain and copy SCons to the folder:
+echo %BIN_DIR%
+exit /b 1
 )
 
-set ARCHITECTURE=%1
-set BUILD_TYPE=%2
+if NOT EXIST "%QTDIR%" (
+echo.
+echo Could not find Qt %QT_VERSION% at "%QT_DIR%".
+exit /b 1
+)
 
-if "%ARCHITECTURE%" == "x86" set res32=T
-if "%ARCHITECTURE%" == "x32" set res32=T
+REM Everyting prepared. Setup the compiler.
+if %MACHINE_X86% (
+call "%BUILDTOOLS_PATH%\%BUILDTOOLS_SCRIPT%" %COMPILER_X86%
+) else (
+call "%BUILDTOOLS_PATH%\%BUILDTOOLS_SCRIPT%" %COMPILER_X64%
+)
 
+REM Now build Mixxx.
 
-if "%res32%" == "T" (
-  set TARGET_MACHINE=x86
-  set VCVARS_ARCH=x86
-  set DISTDIR=dist32
-  if "%BUILD_TYPE%" == "release" (
-    set WINLIB_PATH=%WINLIB_PATH32R%
-  ) else (
-    set WINLIB_PATH=%WINLIB_PATH32D%
-  )
-  echo "****************************"
-  echo "** Building 32 bits Mixxx **"
-  echo "****************************"
+if %CONFIG_RELEASE% (
+set BUILD_TYPE=release
 ) else (
-  set TARGET_MACHINE=amd64
-  set VCVARS_ARCH=x86_amd64
-  set DISTDIR=dist64
-  if "%BUILD_TYPE%" == "release" (
-    set WINLIB_PATH=%WINLIB_PATH64R%
-  ) else (
-    set WINLIB_PATH=%WINLIB_PATH64D%
-  )
-  echo "****************************"
-  echo "** Building 64 bits Mixxx **"
-  echo "****************************"
+set BUILD_TYPE=debug
+)
+
+if %MACHINE_X86% (
+set MACHINE_TYPE=x86
+set DISTDIR=dist32
+) else (
+set MACHINE_TYPE=x86_64
+set DISTDIR=dist64
 )
 
 REM Clean up after old builds.
 REM del /q /f *.exe *.msi 2>NUL
 REM rmdir /s /q %DISTDIR%
 
-call "c:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\vcvarsall.bat" %VCVARS_ARCH%
+rem /MP Use all CPU cores.
+rem /FS force synchronous PDB writes (prevents PDB corruption with /MP)
+rem /EHsc Do not handle SEH in try / except blocks.
+rem /Zc:threadSafeInit- disable C++11 magic static support (Bug #1653368)
+set CXXFLAGS=/MP /FS /EHsc /Zc:threadSafeInit-
+set CFLAGS=/MP /FS /EHsc /Zc:threadSafeInit-
+
+set OLDPATH=%PATH%
+set PATH=%BIN_DIR%;%PATH%
+scons.py mixxx makerelease toolchain=msvs winlib=%WINLIB_DIR% build=%BUILD_TYPE% staticlibs=1 staticqt=1 debug_assertions_fatal=1 verbose=1 machine=%MACHINE_TYPE% qtdir=%QTDIR% hss1394=1 mediafoundation=1 opus=1 localecompare=1 optimize=portable virtualize=0 test=1 qt_sqlite_plugin=0 mssdk_dir="%MSSDK_DIR%" build_number_in_title_bar=0 bundle_pdbs=1
+set PATH=%OLDPATH%
 
-REM set multiprocessor to build faster (together with scons -j4)
-set CL=/MP /FS /EHsc
+REM For debugging, print the configuration log.
+type config.log
```

Comment:
Yea this was just for debugging why configuration was failing. I was going to remove it, but moving it to the failure block is an even better idea